commit aa2d256782e57f9dc5a4421cef1d01444bb7fbb1
Author: Rafael Espindola <rafael.espindola@gmail.com>
Date:   2018-04-02

    Align stubs for external and common global variables to pointer size.
    
    This patch fixes PR36885: clang++ generates unaligned stub symbol
    holding a pointer.
    
    Patch by Rahul Chaudhry!
    
    git-svn-id: https://llvm.org/svn/llvm-project/llvm/trunk@329030 91177308-0d34-0410-b5e6-96231b3b80d8

--- lib/CodeGen/AsmPrinter/AsmPrinter.cpp
+++ lib/CodeGen/AsmPrinter/AsmPrinter.cpp
@@ -1336,6 +1336,7 @@ bool AsmPrinter::doFinalization(Module &M) {
       OutStreamer->SwitchSection(TLOF.getDataSection());
       const DataLayout &DL = M.getDataLayout();
 
+      EmitAlignment(Log2_32(DL.getPointerSize()));
       for (const auto &Stub : Stubs) {
         OutStreamer->EmitLabel(Stub.first);
         OutStreamer->EmitSymbolValue(Stub.second.getPointer(),
